### PR TITLE
Fix #145 bad request on GET /api/owners/{id}/pets/{petId}

### LINKED ISSUE #145: Bad Request Error while trying to get pet for a specific owner
The `GET/api/owners/{id}/pets/{petId}` endpoint seems to be returning an incorrect response or handling errors improperly. This issue was encountered while attempting to retrieve specific pet for a given owner.

cURL: 
```bash
curl -X 'GET' \
  'http://localhost:9966/petclinic/api/owners/1/pets/1' \
  -H 'accept: application/json'
```

Response on Swagger-ui: `Error: response status is 400`

I tried to debug code and found out that it comes from `OwnerRestController` :
```java
@RestController
@RequestMapping("/api")
public class OwnerRestController implements OwnersApi {

    // ...
    
    @PreAuthorize("hasRole(@roles.OWNER_ADMIN)")
    @Override
    public ResponseEntity<PetDto> getOwnersPet(Integer ownerId, Integer petId) {
        Owner owner = this.clinicService.findOwnerById(ownerId);
        Pet pet = this.clinicService.findPetById(petId);
        if (owner == null || pet == null) {
            return new ResponseEntity<>(HttpStatus.NOT_FOUND);
        } else {
            if (!pet.getOwner().equals(owner)) {
                return new ResponseEntity<>(HttpStatus.BAD_REQUEST);
            } else {
                return new ResponseEntity<>(petMapper.toPetDto(pet), HttpStatus.OK);
            }
        }
    }
}
```
The problem is here : `if (!pet.getOwner().equals(owner)) {}`: the `equals()` check returns `false`. Ive checked the entity Owner class and found out that it doesn't have `equals` and `hashcode()` (same for Pet entity class), which is not recommended !

### PR DESCRIPTION
**Pull request by arey**

### CODE REVIEW

File: src/main/java/org/springframework/samples/petclinic/rest/controller/OwnerRestController.java [hunk 1]
```diff
@@ -166,15 +167,13 @@ public ResponseEntity<VisitDto> addVisitToOwner(Integer ownerId, Integer petId,
...
+        if (owner == null) {
             return new ResponseEntity<>(HttpStatus.NOT_FOUND);
-        } else {
-            if (!pet.getOwner().equals(owner)) {
-                return new ResponseEntity<>(HttpStatus.BAD_REQUEST);
-            } else {
-                return new ResponseEntity<>(petMapper.toPetDto(pet), HttpStatus.OK);
-            }
         }
+        Optional<Pet> pet = owner.getPets().stream().filter(p -> p.getId().equals(petId)).findFirst();
```
Comment (firasrg) [inline src/main/java/org/springframework/samples/petclinic/rest/controller/OwnerRestController.java:173]: IMO it's better to let logic goes in Service class, so controllers are focused on endpoint input/output. So i suggest to do : 
```java
Optional<Pet> target = this.clinicService.getPetByOwner(owner, petId);
```

File: src/main/java/org/springframework/samples/petclinic/rest/controller/OwnerRestController.java [hunk 2]
```diff
@@ -166,15 +167,13 @@ public ResponseEntity<VisitDto> addVisitToOwner(Integer ownerId, Integer petId,
...
+        if (pet.isPresent()) {
+            return new ResponseEntity<>(petMapper.toPetDto(pet.get()), HttpStatus.OK);
+        }
+        return new ResponseEntity<>(HttpStatus.BAD_REQUEST);
```
Comment (firasrg) [inline src/main/java/org/springframework/samples/petclinic/rest/controller/OwnerRestController.java:177]: I'd say that NOT_FOUND status would be more suitable here. BAD_REQUEST is when a param is missing or when http method is wrong

File: src/main/java/org/springframework/samples/petclinic/rest/controller/OwnerRestController.java (hunk 1 above)
Comment (arey) [reply src/main/java/org/springframework/samples/petclinic/rest/controller/OwnerRestController.java]: A better place is the `Owner` class ;)

Comment (firasrg) [reply src/main/java/org/springframework/samples/petclinic/rest/controller/OwnerRestController.java]: alright

**Pull request merged**